eZ/Publish/API/Repository/Tests/ContentServiceTest.php

```diff
@@ -5158,6 +5157,123 @@ public function testPublishContentWithoutPublishPolicyThrowsException()
     }
 
     /**
+     * Test removal of the specific translation from all the Versions of a Content Object.
+     *
+     * @covers \eZ\Publish\Core\Repository\ContentService::removeTranslation
+     */
+    public function testRemoveTranslation()
+    {
+        $repository = $this->getRepository();
+        $contentService = $repository->getContentService();
+        $content = $this->createContentVersion2();
+
+        // create multiple versions to exceed archive limit
+        for ($i = 0; $i < 5; ++$i) {
+            $contentDraft = $contentService->createContentDraft($content->contentInfo);
+            $contentUpdateStruct = $contentService->newContentUpdateStruct();
+            $contentDraft = $contentService->updateContent(
+                $contentDraft->versionInfo,
+                $contentUpdateStruct
+            );
+            $contentService->publishVersion($contentDraft->versionInfo);
+        }
+
+        $contentService->removeTranslation($content->contentInfo, 'eng-GB');
+
+        foreach ($contentService->loadVersions($content->contentInfo) as $versionInfo) {
+            self::assertNotContains('eng-GB', $versionInfo->languageCodes);
+            self::assertNotContains('eng-GB', array_keys($versionInfo->getNames()));
+        }
+    }
+
+    /**
+     * Test removal of a main translation throws InvalidArgumentException.
+     *
+     * @covers \eZ\Publish\Core\Repository\ContentService::removeTranslation
+     * @expectedException \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @expectedExceptionMessage Specified translation is the main translation of the Content Object
+     */
+    public function testRemoveMainTranslationThrowsInvalidArgumentException()
+    {
+        $repository = $this->getRepository();
+        $contentService = $repository->getContentService();
+        $content = $this->createContentVersion2();
+
+        // delete first version which has only one translation
+        $contentService->deleteVersion($contentService->loadVersionInfo($content->contentInfo, 1));
```

Comment:
Note: As mentioned above, this doesn't change anything - I just want to reflect separate use cases.